src/printer.js

```diff
@@ -4144,9 +4142,9 @@ function nodeStr(node, options, isFlowOrTypeScriptDirectiveLiteral) {
     canChangeDirectiveQuotes = true;
   }
 
-  const enclosingQuote = shouldUseAlternateQuote
-    ? alternate.quote
-    : preferred.quote;
+  const enclosingQuote = node.__prettier__isJson
```

Comment:
What about options.parser === "json" instead of passing this value all the way through?